core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java

```diff
@@ -42,18 +43,31 @@
   private final TableId tableId; // 2a
   private final String tabletDir; // t-0003
   private final String fileName; // C0004.rf
-  private final String metadataEntry;
   private final Path metaPath;
   private final String normalizedPath;
 
+  private Optional<String> metadataEntry;
+
   /**
-   * Construct a tablet file using a Path object already created. Used in the case where we had to
-   * use Path object to qualify an absolute path.
+   * Construct new tablet file using a Path. Used in the case where we had to use Path object to
+   * qualify an absolute path or create a new file.
    */
-  public TabletFile(Path metaPath, String originalMetaEntry) {
-    this.metadataEntry = Objects.requireNonNull(originalMetaEntry);
+  public TabletFile(Path metaPath) {
+    this(metaPath, Optional.empty());
+  }
+
+  /**
+   * Construct a tablet file using the string read from the metadata. Preserve the exact string so
+   * the entry can be deleted.
+   */
+  public TabletFile(String metadataEntry) {
+    this(new Path(metadataEntry), Optional.of(Objects.requireNonNull(metadataEntry)));
```

Comment:
I think Optional.of will throw an exception if null, so could drop the reqNonNull.

```suggestion
    this(new Path(metadataEntry), Optional.of(metadataEntry));
```